### PR TITLE
Switch to new iris >= 3.10.0 API

### PR DESCRIPTION
**Pull request by schlunma**
## Description

Iris introduces a couple new features and also some breaking changes in [v3.10.0](https://scitools-iris.readthedocs.io/en/v3.10.0rc0/whatsnew/index.html):
- `iris.experimental.ugrid` is deprecated in favor for `iris.mesh`
- `Mesh` is now called `MeshXY`
- Keyword arguments for `MeshXY.coord()` have changed
- `broadcast_to_shape` now supports `chunk` argument (we can remove our custom solution)
- `area_weights` calculation is now lazy (we can remove our custom solution)

I opted for pinning iris instead of making it backwards-compatible so we can use all new features, which allows us to remove lots of lines in our code.

Addresses https://github.com/ESMValGroup/ESMValCore/issues/2492

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
excellent, many thanks for taking on this one, Manu! Lemme know when you need a review :beer:

File: environment.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: esmvaltool
 channels:
   - conda-forge
   - nodefaults
+  - conda-forge/label/rc_iris  # REMOVE BEFORE MERGING THIS PR!
```
Comment (valeriupredoi) [inline environment.yml:6]:
```suggestion
```

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
just adding a placeholder review suggestion so we don't end up with that iris RC channel in `main` :grin:

File: environment.yml [hunk 2]
```diff
@@ -18,7 +19,7 @@ dependencies:
   - fire
   - geopy
   - humanfriendly
-  - iris >=3.9.0
+  - iris >=3.10.0rc0
```
Comment (valeriupredoi) [inline environment.yml:22]:
```suggestion
  - iris >=3.10.0
```

**Review by valeriupredoi — COMMENTED** [review]
remember to pin it in setup.py too

**Comment by schlunma** [comment]
Since iris released [v.3.10.0](https://github.com/SciTools/iris/releases/tag/v3.10.0) today, this is ready for review now!

File: esmvalcore/preprocessor/_shared.py [hunk 3]
```diff
@@ -318,46 +319,6 @@ def get_array_module(*args):
...
-    elif _is_lazy_masked_data(array):
-        # broadcast_to strips masks so we need to handle them explicitly.
-        mask = da.ma.getmaskarray(array)
-        new_mask = broadcast(mask)
-        new_array = da.ma.masked_array(new_array, new_mask)
-
-    return new_array
-
-
 def _is_lazy_masked_data(array):
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_shared.py:322]: This function can also be removed, it was only used by the function above

Comment (schlunma) [reply esmvalcore/preprocessor/_shared.py]: Done in https://github.com/ESMValGroup/ESMValCore/pull/2500/commits/3a7a1804fc31d3ec1ab9f2021d0b991c9887b556

**Comment by valeriupredoi** [comment]
will do a more reviewey review tomorrow, in the meantime, be advised of this for future `rc_iris` stuffs https://github.com/conda-forge/iris-feedstock/issues/115

**Review by valeriupredoi — APPROVED** [review]
no problems at all from my side, very nice and sound PR, Manu :beer: Only thing we have to be careful is if anyone develops a diagnostic using the `main` of ESMValCore (so in full development way) then wants it deployed in Tool - we'll have to tell them to wait for a Core release, but let's bridge that bridge if we get there

**Comment by schlunma** [comment]
@bouweandela please merge if you're happy with this 👍

**Comment by schlunma** [comment]
Thanks V for reviewing 🍻

**Pull request merged**